src/components/call/CreateUpdateCall.tsx

```diff
@@ -106,15 +72,47 @@ const CreateUpdateCall: React.FC<CreateUpdateCallProps> = ({ call, close }) => {
   const currentDayEnd = new Date();
   currentDayEnd.setHours(23, 59, 59, 999);
 
-  const handleNext = () => {
-    setActiveStep(prevActiveStep => prevActiveStep + 1);
+  const steps = ['General', 'Reviews', 'Notification and cycle'];
+
+  const getStepContent = (step: number) => {
+    switch (step) {
+      case 0:
+        return (
+          <CallGeneralInfo
+            templates={templates}
+            loadingTemplates={loadingTemplates}
+            proposalWorkflows={proposalWorkflows}
+            loadingProposalWorkflows={loadingProposalWorkflows}
+          />
+        );
+      case 1:
+        return <CallReviewsInfo />;
+      case 2:
+        return <CallNotificationAndCycleInfo />;
+      default:
+        return 'Unknown step';
+    }
   };
 
-  const handleBack = () => {
+  const handleNext = (formProps: FormikProps<Call>) => {
+    formProps.submitForm();
+
+    if (formProps.isValid) {
+      formProps.validateForm();
+      formProps.setTouched({});
+      setActiveStep(prevActiveStep => prevActiveStep + 1);
+    }
+  };
+
+  const handleBack = (formProps: FormikProps<Call>) => {
+    formProps.setTouched({});
+
     setActiveStep(prevActiveStep => prevActiveStep - 1);
   };
 
-  const handleStep = (step: number) => () => {
+  const handleStep = (step: number, formProps: FormikProps<Call>) => () => {
+    formProps.setTouched({});
```

Comment:
What does the setTouched() do in this case?

Reply:
It was used in some examples that I have seen in formik github repo. It is mainly because when you change from one to another step and try to check `formProps.isValid` is always initially `false` even if you have all the valid fields. This is like a trigger to just to change the state if everything is valid inside selected step.